src/main/java/com/github/mizool/jediscache/JedisCachingProvider.java

```diff
@@ -0,0 +1,193 @@
+package com.github.mizool.jediscache;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.WeakHashMap;
+
+import javax.cache.CacheException;
+import javax.cache.CacheManager;
+import javax.cache.configuration.OptionalFeature;
+import javax.cache.spi.CachingProvider;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.kohsuke.MetaInfServices;
+
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.JedisPoolConfig;
+
+@Slf4j
+@MetaInfServices(CachingProvider.class)
+public class JedisCachingProvider implements CachingProvider
+{
+    private static final String JEDISCACHE_HOST = "jediscache.host";
+    private final JedisPool jedisPool;
+    private Map<ClassLoader, Map<URI, CacheManager>> cacheManagersByClassLoader;
+
+    public JedisCachingProvider()
+    {
+        String host = System.getProperty(JEDISCACHE_HOST, "127.0.0.1");
+        log.info("JedisCache active, connecting to {}", host);
+        jedisPool = new JedisPool(new JedisPoolConfig(), host);
+        cacheManagersByClassLoader = new WeakHashMap<>();
+    }
+
+    @Override
+    public synchronized CacheManager getCacheManager(URI uri, ClassLoader classLoader, Properties properties)
+    {
+        if (uri == null)
+        {
+            uri = getDefaultURI();
+        }
+        if (classLoader == null)
+        {
+            classLoader = getDefaultClassLoader();
+        }
+        if (properties == null)
+        {
+            properties = new Properties();
+        }
+
+        Map<URI, CacheManager> cacheManagersByURI = cacheManagersByClassLoader.get(classLoader);
```

Comment:
extract this and the `if` below into `getOrCreateCacheManagersByUri`

also, in all names that contain `URI`, change that to `Uri`